File: backend/backend/domain/repositories/material_repository.py
"""MaterialRepository 인터페이스."""

from __future__ import annotations

from abc import abstractmethod

from backend.domain.entities.material import Material, MaterialType
from backend.domain.repositories.base import BaseRepository


class MaterialRepository(BaseRepository[Material, str]):
    """자재 Repository 인터페이스."""

    @abstractmethod
    async def get_by_type(self, material_type: MaterialType) -> list[Material]:
        """유형별 자재 조회."""
        ...

File: frontend/src/domain/repositories/material.repository.ts
/**
 * Material Repository Interface
 * 자재 데이터 접근 인터페이스
 */

import type {
  Material,
  MaterialCreateInput,
  MaterialUpdateInput,
  MaterialType,
} from '../entities/material';
import type { BaseRepository } from './base.repository';

export interface IMaterialRepository
  extends BaseRepository<Material, MaterialCreateInput, MaterialUpdateInput> {
  /**
   * 자재 유형별 조회
   */
  getByType(type: MaterialType): Promise<Material[]>;

  /**
   * 자재 일괄 수정
   */
  bulkUpdate(materials: Array<{ id: string } & MaterialUpdateInput>): Promise<Material[]>;

  /**
   * 자재 검색 (품번, 품명)
   */
  search(query: string): Promise<Material[]>;
}
